pkg/build/builder/sti.go

```diff
@@ -204,19 +215,42 @@ func (s *S2IBuilder) Build() error {
 		return err
 	}
 
+	// TODO: why in docker builds we defer removing the image, while here we don't?!
```

Comment:
@bparees @mfojtik may I ask you why we don't defer removing the built image in Docker builds, while we do it in S2I?

Feature or bug?